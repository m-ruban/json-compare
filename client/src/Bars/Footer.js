import React from 'react'
import { AppConsumer } from '../Contexts/Provider'

const Footer = () => (
  <footer className="bg-light">
    <AppConsumer>
      {
        ({ autor }) => (
          <div className="autor text-muted py-3">Autor -&nbsp;
            <a target="_blank" href={autor} rel="noopener noreferrer">DD</a>
          </div>
        )
      }
    </AppConsumer>
  </footer>
)

export default Footer
