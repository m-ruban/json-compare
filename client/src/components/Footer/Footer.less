@import "~styles/variables.less";

.footer {
  background-color: @secondary-background-color !important;

  &__container {
    border-top: 1px solid @border-color;
  }

  &__author {
    padding: 0.5rem 1rem;
    font-size: 14px;
  }

  &__link {
    color: @secondary-color;
  }
}
