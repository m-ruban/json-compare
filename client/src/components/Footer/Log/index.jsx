import React from 'react';
import PropTypes from 'prop-types';
import withConsumer from 'hoc/withConsumer';
import 'components/Footer/Log/Log.less';

const Log = ({ log }) => {
    if (!log.type) {
        return null;
    }

    return (
        <div className="log pr-3 py-3">
            <div className="log__item">
                <div className={`log__${log.type}`}>
                    <span className="log__type">log.{log.type}</span>: {log.text}
                </div>
            </div>
        </div>
    );
};

Log.propTypes = {
    log: PropTypes.shape({
        type: PropTypes.string,
        text: PropTypes.text,
    }),
};

export default withConsumer(Log);
