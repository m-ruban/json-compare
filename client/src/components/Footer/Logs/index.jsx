import React from 'react';
import PropTypes from 'prop-types';
import withConsumer from 'hoc/withConsumer';

const Logs = ({ log }) => {
    if (!log.type) {
        return null;
    }

    return (
        <div className="logs pr-3 py-3">
            <div className="list">
                <span className={log.type}>
                    <u>log.{log.type}</u>: {log.text}
                </span>
            </div>
        </div>
    );
};

Logs.propTypes = {
    log: PropTypes.shape({
        type: PropTypes.string,
        text: PropTypes.text,
    }),
};

export default withConsumer(Logs);
