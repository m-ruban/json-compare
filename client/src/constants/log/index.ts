export const LOG_INFO = 'info';
export const LOG_ERROR = 'error';
