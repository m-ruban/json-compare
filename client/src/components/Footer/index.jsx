import React from 'react';
import PropTypes from 'prop-types';
import { Row, Container } from 'reactstrap';

import withConsumer from 'hoc/withConsumer';
import trl from 'modules/translation';
import Log from 'components/Footer/Log';
import 'components/Footer/Footer.less';

const Footer = ({ author }) => (
    <footer className="bg-light footer">
        <Container className="footer__container" fluid>
            <Row>
                <div className="footer__author text-muted py-3">
                    {trl('Footer.text')}
                    <a className="footer__link" target="_blank" href={author} rel="noopener noreferrer">
                        {trl('Footer.name')}
                    </a>
                </div>
                <Log />
            </Row>
        </Container>
    </footer>
);

Footer.propTypes = {
    author: PropTypes.string,
};

export default withConsumer(Footer);
