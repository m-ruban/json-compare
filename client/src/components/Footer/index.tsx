import React from 'react';
import { Container, Row } from 'reactstrap';

import Log from 'components/Footer/Log';
import withConsumer from 'hoc/withConsumer';
import trl from 'modules/translation';

import 'components/Footer/Footer.less';

interface IFooterProps {
    author: string;
}

const Footer = ({ author }: IFooterProps) => (
    <footer className="footer">
        <Container className="footer__container" fluid>
            <Row>
                <div className="footer__author text-muted py-3">
                    {trl('Footer.text')}
                    <a className="footer__link" target="_blank" href={author} rel="noopener noreferrer">
                        {trl('Footer.name')}
                    </a>
                </div>
                <Log />
            </Row>
        </Container>
    </footer>
);

export default withConsumer(Footer);
