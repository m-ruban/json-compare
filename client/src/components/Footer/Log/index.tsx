import { ILog } from 'constants/interfaces';
import React from 'react';

import withConsumer from 'hoc/withConsumer';

import 'components/Footer/Log/Log.less';

const Log = ({ log }: { log: ILog }) => {
    if (!log.type) {
        return null;
    }
    return (
        <div className="log pr-3 py-3">
            <div className="log__item">
                <div className={`log__${log.type}`}>
                    <span className="log__type">log.{log.type}</span>: {log.text}
                </div>
            </div>
        </div>
    );
};

export default withConsumer(Log);
