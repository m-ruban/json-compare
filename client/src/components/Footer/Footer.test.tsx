import React from 'react';

import { render } from 'enzyme';

import Footer from 'components/Footer';
import { Provider } from 'store/Context';
import feedback from 'tests/mocks/feedback';
import { emptyLog } from 'tests/mocks/log';

const initialState = {
    ...emptyLog,
    ...feedback,
};

describe('<Footer />', () => {
    it('Renders correctly', () => {
        const output = render(
            <Provider value={initialState}>
                <Footer />
            </Provider>,
        );
        expect(output).toMatchSnapshot();
    });
});
