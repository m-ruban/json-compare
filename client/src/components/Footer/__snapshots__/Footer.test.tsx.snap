// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Renders correctly 1`] = `
<footer
  class="footer"
>
  <div
    class="footer__container container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="footer__author text-muted py-3"
      >
        Author - 
        <a
          class="footer__link"
          href="test_link"
          rel="noopener noreferrer"
          target="_blank"
        >
          wardak
        </a>
      </div>
    </div>
  </div>
</footer>
`;
